ipie/addons/thermal/utils/ueg.py

```diff
@@ -0,0 +1,583 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+import numpy
+import scipy.sparse
+from ipie.utils.io import write_qmcpack_sparse
+
+# -----------------------------------------------------------------------------
+# System class.
+# -----------------------------------------------------------------------------
+class UEG(object):
+    """UEG system class (integrals read from fcidump)
+
+    Parameters
+    ----------
+    nup : int
+        Number of up electrons.
+
+    ndown : int
+        Number of down electrons.
+
+    rs : float
+        Density parameter.
+
+    ecut : float
+        Scaled cutoff energy.
+
+    ktwist : :class:`numpy.ndarray`
+        Twist vector.
+
+    verbose : bool
+        Print extra information.
+
+    Attributes
+    ----------
+    T : :class:`numpy.ndarray`
+        One-body part of the Hamiltonian. This is diagonal in plane wave basis.
+
+    ecore : float
+        Madelung contribution to the total energy.
+
+    h1e_mod : :class:`numpy.ndarray`
+        Modified one-body Hamiltonian.
+
+    nfields : int
+        Number of field configurations per walker for back propagation.
+
+    basis : :class:`numpy.ndarray`
+        Basis vectors within a cutoff.
+
+    kfac : float
+        Scale factor (2pi/L).
+    """
+
+    def __init__(self, options, verbose=False):
```

Comment:
Again not your problem but we shouldn't pass in big dicts of parameters. Feel free to not address this now.